plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_incident_events/task.py

```diff
@@ -0,0 +1,91 @@
+import insightconnect_plugin_runtime
+from .schema import (
+    MonitorIncidentEventsInput,
+    MonitorIncidentEventsOutput,
+    MonitorIncidentEventsState,
+    Input,
+    Output,
+    Component,
+    State,
+)
+
+# Custom imports below
+import json
+
+from ...util.util import Util
+from typing import List, Dict
+
+
+INITIAL_HISTORICAL_LOOK_BACK_MS = 1 * 60 * 60 * 1000
+DEFAULT_TIME_SORTING_FIELD = "modification_time"
+
+
+class MonitorIncidentEvents(insightconnect_plugin_runtime.Task):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="monitor_incident_events",
+            description=Component.DESCRIPTION,
+            input=MonitorIncidentEventsInput(),
+            output=MonitorIncidentEventsOutput(),
+            state=MonitorIncidentEventsState(),
+        )
+
+    @staticmethod
+    def get_request_filters(status: str, descriptions: List[str], incident_ids: List[str]) -> List[Dict]:
+        filters = []
+        if status is not None and len(status) > 0:
+            filters.append({"field": "status", "operator": "eq", "value": status})
+
+        if descriptions is not None and len(descriptions) > 0:
+            filters.append({"field": "description", "operator": "in", "value": descriptions})
+
+        if incident_ids is not None and len(incident_ids) > 0:
+            filters.append({"field": "incident_id_list", "operator": "in", "value": incident_ids})
```

Comment:
```suggestion
        if status:
            filters.append({"field": "status", "operator": "eq", "value": status})

        if descriptions:
            filters.append({"field": "description", "operator": "in", "value": descriptions})

        if incident_ids:
            filters.append({"field": "incident_id_list", "operator": "in", "value": incident_ids})
```

Optional, but the expressions can be simplified